src/main.rs

```diff
@@ -81,89 +29,9 @@ Options:
 fn main() {
     let config = Config::default().unwrap();
     let args = env::args().collect::<Vec<_>>();
-    let result = cargo::call_main_without_stdin(real_main, &config, USAGE, &args, false);
+    let result = cargo::call_main_without_stdin(cargo_ebuild::real_main, &config, USAGE, &args, false);
```

Comment:
This is just wrapping a `main()` function into the library. Let's work out a real API and have `lib.rs` expose that API and then switch the binary to using that. A better way to structure the changes would be commits that work in the following order:
- add new API
- implement & test
- use new API